### DISCUSSION TITLE
Ultralytics Web API (Fast Api or Flask)

### DISCUSSION THREAD

**Discussion by Klinefelters** [ultralytics/ultralytics#13261] [Ideas]
My feature idea is a backend api to interface with ultralytics from a web application. I was also considering building an example web application that could interface with ultralytics and make changes in real time to a running model. The idea would be to allow the user to change models, control variables, and possibly camera settings as well.

In the past year or so I've been building up experience working with web applications, and I was wondering if this idea is one worth pursuing and developing as a personal project. Any thoughts and recommendations are appreciated.

**Comment by pderrenger** [comment]
Hello!

Your idea of developing a backend API to interface with Ultralytics models through a web application sounds fantastic and very promising! Building real-time capabilities for model interaction and control could significantly enhance user experience and functionality.

Given your experience with web applications, this project could serve as an excellent showcase of integrating cutting-edge AI models with web technologies. It's definitely a worthwhile pursuit, and it could also contribute valuable insights and tools to the community.

If you decide to move forward with this, documenting your progress and sharing your findings could be incredibly beneficial for others looking to explore similar integrations.

Best of luck with your project, and we're excited to see what you'll create!
